Emby.Server.Implementations/HttpServer/HttpListenerHost.cs

```diff
@@ -530,22 +535,25 @@ public async Task RequestHandler(IHttpRequest httpReq, string urlString, string
                 }
                 else
                 {
-                    await ErrorHandler(new FileNotFoundException(), httpReq, false).ConfigureAwait(false);
+                    throw new FileNotFoundException();
                 }
             }
-            catch (Exception ex) when (ex is SocketException || ex is IOException || ex is OperationCanceledException)
-            {
-                await ErrorHandler(ex, httpReq, false).ConfigureAwait(false);
-            }
-            catch (SecurityException ex)
-            {
-                await ErrorHandler(ex, httpReq, false).ConfigureAwait(false);
-            }
             catch (Exception ex)
             {
-                var logException = !string.Equals(ex.GetType().Name, "SocketException", StringComparison.OrdinalIgnoreCase);
+                // Do not handle exceptions manually when in development mode
+                // The framework-defined development exception page will be returned instead
+                if (_hostEnvironment.IsDevelopment())
+                {
+                    throw;
+                }
 
-                await ErrorHandler(ex, httpReq, logException).ConfigureAwait(false);
+                bool ignoreStackTrace =
+                    ex is SocketException ||
+                    ex is IOException ||
+                    ex is OperationCanceledException ||
+                    ex is SecurityException ||
+                    ex is FileNotFoundException;
```

Comment:
Weren't we doing boolean operators at the beginning?

Reply:
Yes we are, good catch